### PR TITLE
docs:#823 Clarify encryption docs

### LINKED ISSUE #823: Clarify mTLS documentation
### Describe the Issue

feedback on the lounge session that the mTLS docs were hard to use and they were not able to get it working based upon it. See
https://redpandadata.slack.com/archives/C01LK3UK2K1/p1668531596356479
https://redpandacommunity.slack.com/archives/C048589SBEZ/p1668529728609809

### Updates to existing documentation

Here's where our doc covers mTLS. Joe's working on the K8s page (see [here](https://redpandacommunity.slack.com/archives/C048589SBEZ/p1668529728609809)), so please look at the encryption & authentication pages first!
[Configuring Encryption](https://docs.redpanda.com/docs/platform/security/encryption/#mtls)
[Configuring Authentication](https://docs.redpanda.com/docs/platform/security/authentication/)
[Configuring mTLS on Kubernetes](https://docs.redpanda.com/docs/platform/reference/redpanda-operator/kubernetes-mtls/) (reference, with Redpanda Operator)

### LINKED ISSUE #1045: cert script should be moved out of personal gist
### Describe the Issue

Encryption page has a link to a Juan's personal gist: https://gist.github.com/0x5d/30404e0b3376a3724da828f1db5a3b1d29ffebf5

Script should be moved somewhere else. 

### Updates to existing documentation

https://docs.redpanda.com/docs/manage/security/encryption/ 

### Additional notes

Juan is asking an admin to create a new gist for the script.

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #823 
resolves #1045 

This replaces the encryption prereq (which linked to Juan's personal gist and had errors) with updated steps from Denis.

preview: https://deploy-preview-1148--redpanda-documentation.netlify.app/docs/manage/security/encryption/

### CODE REVIEW

**Comment by micheleRP** [comment]
Hi @dcoded and @patrickangeles! Please review the updated Prerequisites section for TLS/mTLS, and please add anyone else you think should review it.

File: docs/manage/security/encryption.mdx [hunk 1]
```diff
@@ -4,22 +4,148 @@ title: Configure Encryption
...
+## Prerequisites
 
-TLS certificates are necessary for encryption. You can use your own certificates, either self-signed or issued by a trusted Certificate Authority, or you can use use the Redpanda [prepared script](https://gist.github.com/0x5d/30404e0b3376a3724da828f1db5a3b1d29ffebf5) to generate certificates:
+TLS certificates are necessary for encryption. You can use your own certificates, either self-signed or issued by a trusted certificate authority (CA), or you can follow these steps.
+
+### Create certificate signing requests
+
+#### Create node certificate configuration
+
+A CNF file may need to be made for each node to modify the `alt_names` section. For production usage, update the `distinguished_name` section.
```
Comment (Feediver1) [inline docs/manage/security/encryption.mdx:22]:
```suggestion
You might need to create a CNF file for each node to modify the `alt_names` section. For production usage, update the `distinguished_name` section.
```

File: docs/manage/security/encryption.mdx [hunk 2]
```diff
@@ -4,22 +4,148 @@ title: Configure Encryption
...
+#   -config node.cnf Configuration file to use when generating certificates (created above).
+#   -key ca.key      Private key of the node (created above).
+#
+# Outputs:
+#   -out node.crt    The public key certificate of the CA.
+#
+openssl req -new -config node.cnf -key node.key -out node.csr
+```
+
+If done by a corporate CA, these certificate signing requests must be signed with the following extensions:
```
Comment (Feediver1) [inline docs/manage/security/encryption.mdx:72]:
```suggestion
If generated by a corporate CA, these certificate signing requests must be signed with the following extensions:
```

**Review by Feediver1 — APPROVED** [review]
LGTM

**Comment by micheleRP** [comment]
@dcoded, @patrickangeles: Could you approve this?

**Comment by micheleRP** [comment]
status = Denis ran into issues testing this. Waiting for Patrick to have time to test.

**Comment by patrickangeles** [comment]
Some comments:
- Maybe better titled as **Configuring Kafka TLS Encryption** (so as not to be confused with on-disk encryption or something else)
- This covers client-to-broker communications via the Kafka API... I haven't seen how you setup TLS for broker-to-broker RPC communications. Is that covered in another section?
- Some sections could use further context. For example: why is this step optional? **Optional: Create a local CA for self-signed certificates**
- Because of `# comments` for the `openssl` commands, copy/paste doesn't work. Suggestion: break out the comments so that they're in normal document text, not part of the copy/paste block (those should be just for the commands).

General comment: overall it reads somewhere between a HOWTO and a reference doc... it might be clearer to break these out into separate pages? Or structure this page somehow so that we go through what needs to be configured, but also show some of the example steps but be clear that they're examples.

Related is that the examples seems to assume a host based installation... you can't just edit `redpanda.yaml` for a container based install. So the examples won't apply to any user trying this on a docker based setup (which is a likely scenario if you're following a HOWTO and trying this on your local machine).

File: docs/manage/security/encryption.mdx [hunk 3]
```diff
@@ -1,27 +1,167 @@
...
+openssl req -new -config node.cnf -key node.key -out node.csr
+```
+
+where:
+
+| Inputs              | Description   |
+| --------            | ------    | 
+| `-new`                | New request.    | 
+| `-config node.cnf`    | Configuration file to use when generating certificates (created above).    | 
+| `-key ca.key`         | Private key of the node (created above).    |
```
Comment (micheleRP) [inline docs/manage/security/encryption.mdx:70]:
```suggestion
| `-key node.key`         | Private key of the node (created above).    | 
```
@patrickangeles the code above uses `-key node.key`. Should I change this to match?

File: docs/manage/security/encryption.mdx [hunk 4]
```diff
@@ -1,27 +1,167 @@
...
+
+| Inputs              | Description   |
+| --------            | ------    | 
+| `-new`                | New request.    | 
+| `-config node.cnf`    | Configuration file to use when generating certificates (created above).    | 
+| `-key ca.key`         | Private key of the node (created above).    | 
+
+| Output             | Description   |
+| --------            | ------    | 
+| `-out node.crt`     | The public key certificate of the CA.    |
```
Comment (micheleRP) [inline docs/manage/security/encryption.mdx:74]:
```suggestion
| `-out node.csr`     | The public key certificate of the CA.    |  
```
@patrickangeles the code above uses `-out node.csr`. Should I change this to match?

**Comment by micheleRP** [comment]
@patrickangeles, @dcoded: I renamed this as Patrick suggested, I moved introductory info to the top of the page, and I removed comments from the examples so they copy/paste better. Examples now includes <editable-variables> in red, so you can replace them with actual values before you copy it. (If these don't appear red/editable, the page somehow refreshed, and just reload it by toggling to another page and back to see them editable again.)

I added a couple questions. The remaining feedback is good and I don't want to lose it... but I also don't want to hold up the edits on this page from publishing. Unless either of you have can provide more content for the items below, I'd like to merge this PR and put the below items into a new doc issue to follow up separately. Please let me know what you think!

- This covers client-to-broker communications via the Kafka API... I haven't seen how you setup TLS for broker-to-broker RPC communications. Is that covered in another section? (michele: No. I think the only bit we have is here: https://docs.redpanda.com/docs/reference/node-properties/#rpc_server_tls)
- Some sections could use further context. For example: why is this step optional? Optional: Create a local CA for self-signed certificates (michele: I don't know. Can either of you explain that?)
- General comment: overall it reads somewhere between a HOWTO and a reference doc... it might be clearer to break these out into separate pages? Or structure this page somehow so that we go through what needs to be configured, but also show some of the example steps but be clear that they're examples. (michele: The authentication and authorization pages also should be broken into separate conceptual and task pages.)
- Related is that the examples seems to assume a host based installation... you can't just edit redpanda.yaml for a container based install. So the examples won't apply to any user trying this on a docker based setup (which is a likely scenario if you're following a HOWTO and trying this on your local machine). (michele: This is beyond me, as I've only used redpanda with docker. I'd need an SME to help with this.)

**Comment by micheleRP** [comment]
To wrap up this original issue, I'm moving remaining open comments into new doc issue: https://github.com/redpanda-data/documentation/issues/1425

File: docs/manage/security/encryption.mdx (hunk 4 above)
Comment (dcoded) [reply docs/manage/security/encryption.mdx:80]: Yes, this would be `-out node.csr`.

**Comment by micheleRP** [comment]
merging this, with confirmation from Denis

**Pull request merged**